apis/logging/v1/cluster_log_forwarder_types.go

```diff
@@ -147,11 +147,33 @@ type OutputSpec struct {
 	// Secret for authentication.
 	// Name of a secret in the same namespace as the cluster logging operator.
 	//
-	// For client authentication, set secret keys `tls.crt` and `tls.key` to the client certificate and private key.
+	// All sensitive authentication information is provided via a kubernetes Secret object.
+	// A Secret is a key:value map, common keys are described here.
+	// Some outputs support additional pecialized keys, documented with the output-specific configuration field.
+	// All secret keys are optional, enable the security features you want by setting the relevant keys.
 	//
-	// To use your own certificate authority, set secret key `ca-bundle.crt`.
+	// Transport Layer Security (TLS)
 	//
-	// Depending on the `type` there may be other secret keys that have meaning.
+	// Basic TLS is enabled by TLS URL such as 'https://...' or 'ssl://...', it does not require a Secret.
+	// For basic TLS, the client authenticates the server using the system's default certificate authority.
+	//
+	// Additional TLS features are enabled by selectively setting the following optional fields.
+	//
+	//   `tls.crt`: (string) File name of a client certificate to authenticate the client. Enables mutual authentication. Requires `tls.key`.
+	//   `tls.key`: (string) File name containing the private key  to unlock the client certificate. Requires `tls.crt`
+	//   `passphrase`: (string) Passphrase to decode an encoded private key. Requires tls.key.
+	//   `ca-bundle.crt`: (string) File name of a custom CA for server authentication.
+	//
+	// Username and Password
+	//
+	//   `user`: (string) Authentication user name. `password` must also be set.
```

Comment:
We already are using `username` for the ES password feature that landed.  This means we either: change it here or honor both with one taking priority

Reply:
I don't think that's a problem - in the context of an "elasticsearch" output it means what it means now. In the context of other outputs it may mean something else (basic auth, or user/pass over TLS) or not be applicable at all. The idea is to find a minimal  set of generic keys that can be interpreted appropriately in the context of what a given output supports - and add specific ones (e.g. aws_blah) where an output does something thta doesn't fit the generic pattern. 
Am I missing something?